Entities/Material.h

```diff
@@ -1,376 +1,186 @@
+		void SetIndex(unsigned char newIndex) { m_Index = newIndex; }
+
+		/// <summary>
+		/// Gets the drawing priority of this Material. The higher the number, the higher chances that a pixel of this material will be drawn on top of others.
+		/// </summary>
+		/// <returns>The drawing priority of this Material.</returns>
+		int GetPriority() const { return m_Priority; }
+
+		/// <summary>
+		/// The impulse force that a particle needs to knock loose a terrain pixel of this material. In kg * m/s.
+		/// </summary>
+		/// <returns>Returns impulse force that a particle needs to knock loose a terrain pixel of this material.</returns>
+		float GetIntegrity() const { return m_Integrity; }
+
+		/// <summary>
+		/// Gets the scalar value that defines the restitution of this Material. 1.0 = no kinetic energy is lost in a collision, 0.0 = all energy is lost (plastic).
+		/// </summary>
+		/// <returns>A float scalar value that defines the restitution of this Material.</returns>
+		float GetRestitution() const { return m_Restitution; }
+
+		/// <summary>
+		/// Gets the scalar value that defines the friction of this Material. 1.0 = will snag onto everything, 0.0 = will glide with no friction.
+		/// </summary>
+		/// <returns>A float scalar value that defines the friction of this Material.</returns>
+		float GetFriction() const { return m_Friction; }
+
+		/// <summary>
+		/// Gets the scalar value that defines the stickiness of this Material. 1.0 = will stick to everything, 0.0 = will never stick to anything.
+		/// </summary>
+		/// <returns>A float scalar value that defines the stickiness of this Material.</returns>
+		float GetStickiness() const { return m_Stickiness; }
+
+		/// <summary>
+		/// Gets the density of this Material in Kg/L.
+		/// </summary>
+		/// <returns>The density of this Material.</returns>
+		float GetVolumeDensity() const { return m_VolumeDensity; }
+
+		/// <summary>
+		/// Gets the density of this Material in kg/pixel, usually calculated from the KG per Volume L property.
+		/// </summary>
+		/// <returns>The pixel density of this Material.</returns>
+		float GetPixelDensity() const { return m_PixelDensity; }
+
+		/// <summary>
+		/// If this material transforms into something else when settling into the terrain, this will return that different material index. If not, it will just return the regular index of this material.
+		/// </summary>
+		/// <returns>The settling material index of this or the regular index.</returns>
+		unsigned char GetSettleMaterial() const { return (m_SettleMaterialIndex != 0) ? m_SettleMaterialIndex : m_Index; }
+
+		/// <summary>
+		/// Gets the material index to spawn instead of this one for special effects.
+		/// </summary>
+		/// <returns>The material index to spawn instead of this one for special effects. 0 means to spawn the same material as this.</returns>
+		unsigned char GetSpawnMaterial() const { return m_SpawnMaterialIndex; }
```

Comment:
AFAIK only applicable for terrain.
Basically if the terrain is `MaterialA`, it can turn into `MaterialB` when dislodged.
So like, if you have some sand and you want it to turn to metal for god knows what reason, you have this property.
Or like if you have ice and you pop some caps into it the pixels that get dislodged turn to water.

Reply:
Huh, odd. Yeah, it's also supported as TransformsInto in ini apparently, which is a more self-explanatory name. Well, leave it be and feel free to resolve this.